account-kit/core/src/actions/createAccount.ts

```diff
@@ -154,22 +156,23 @@ export async function createAccount<TAccount extends SupportedAccountTypes>(
         return account;
       });
     } else if (isModularV2AccountParams(params)) {
-      return createModularAccountV2({
-        ...accountConfigs[chain.id]?.[params.type],
-        ...params.accountParams,
-        signer,
-        transport: (opts) => transport({ ...opts, retryCount: 0 }),
-        chain,
-      }).then((account) => {
-        CoreLogger.trackEvent({
-          name: "account_initialized",
-          data: {
-            accountType: "ModularAccountV2",
-            accountVersion: "v2.0.0",
-          },
+      // TODO: we can probably do away with some of the if-else logic here and just convert the params to creation hints
+      // and pass them to the client
+      return smartWalletClient
```

Comment:
This is probably the most beneficial place to add support for wallet apis because it means accounts will be index correctly. everything downstream will the touch this flow which is what we want